pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/realtimetoofflinesegments/RealtimeToOfflineSegmentsTaskGenerator.java

```diff
@@ -179,6 +179,10 @@ public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
               skipGenerate = true;
               break;
             }
+            if (StringUtils.isBlank(segmentZKMetadata.getDownloadUrl())) {
```

Comment:
Doing this for compaction task is fine, but I don't think we can silently swallow exception for this task since it may lead to some data-loss kind of issues for users.

Reply:
Umm should we throw a RuntimeException from here then?

Reply:
removed this change for now as it needs a bigger change in the base classes. Will follow-up in a separate PR.